dev-docs/RFCs/v8.0/ui-module-rfc.md

```diff
@@ -0,0 +1,137 @@
+# UI Module RFC
+
+* **Author**: Xiaoji Chen
+* **Date**: December 2022
+* **Status**: **Pending Review**
+
+## Overview
+
+Visualization applications often require some chrome around the WebGL canvas to offer better user experience. While it is possible for every app to implement their own UI components that interact with deck.gl, it is no trivial effort to implement them well. 
+
+This RFC proposes to add a new module that packages common UI components that seamlessly integrate with deck.gl, including but not limited to:
+
+- Marker: HTML element pinned to a location in the world space
+- Navigation: buttons to zoom / rotate / reset the viewport
+- Fullscreen: button to enter / exit fullscreen mode
+- Geolocate: button to track the user's location using the browser's Geolocation API
+- Legend: visual comparison of sizes, colors etc. corresponding to the rendered layers and viewport
+
+
+## Background
+
+Deck.gl does not currently export any DOM-based UI components. The documentation recommends react-map-gl as a companion library in the following pattern:
+
+```jsx
+import {StaticMap, MapContext, NavigationControl, FullscreenControl} from 'react-map-gl'
+
+<DeckGL
+  ContextProvider={MapContext.Provider}
+>
+  <StaticMap />
+  <NavigationControl />
+  <FullscreenControl />
+</DeckGL>
+```
+
+There are major limitations with this approach:
+
+- It is only supported by react-map-gl v5 and v6, versions no longer actively maintained. v7 dropped deck.gl support to prioritize standalone performance, see https://github.com/visgl/react-map-gl/issues/1646.
+- It pulls in an entire base map library even if the application does not need Mapbox.
+- It is React only, cannot be used in vanilla JS.
+- It is geospatial only, does not work with non-geospatial views.
+- It cannot be used via the declarative interface e.g. with pydeck.
+
+
+## Proposal
+
+1. Export a new interface `UIControl` from `@deck.gl/core`. It will define certain hook methods that a UI component can implement, along the lines of:
+
+```js
+interface UIControl<PropsT> {
```

Comment:
Nit: UIControl seems a bit generic if they are strongly tied to deck.gl. Not sure maybe call them DeckControl?